src/storage-types/src/connections/aws.rs

```diff
@@ -23,52 +23,132 @@ include!(concat!(
     env!("OUT_DIR"),
     "/mz_storage_types.connections.aws.rs"
 ));
+/// The materialize principal details
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub struct MaterializePrincipal {
+    /// The role which Materialize will assume first to eventually
+    /// assume the customer's role via role chaining.
+    pub arn: String,
+    /// The external ID prefix, will need the connection ID as a suffix
+    /// to get the complete External ID to assume customer's role.
+    pub external_id_prefix: String,
+}
+
+impl MaterializePrincipal {
+    pub fn get_external_id(&self, aws_external_id_suffix: String) -> String {
+        format!("mz_{}_{}", self.external_id_prefix, aws_external_id_suffix)
+    }
+
+    pub fn get_aws_example_trust_policy(
+        &self,
+        aws_external_id_suffix: String,
+    ) -> serde_json::Value {
+        serde_json::json!(
+            {
+            "Version": "2012-10-17",
+            "Statement": [
+              {
+                "Effect": "Allow",
+                "Principal": {
+                  "AWS": self.arn
+                },
+                "Action": "sts:AssumeRole",
+                "Condition": {
+                  "StringEquals": {
+                    "sts:ExternalId": self.get_external_id(aws_external_id_suffix)
+                  }
+                }
+              }
+            ]
+          }
+        )
+    }
```

Comment:
...and these can become methods on the `AwsConfig`. I'd drop the `aws` from `get_aws_example_trust_policy` since it'll be implied by `AwsConfig`.

Reply:
I made this part of the `impl AwsAssumeRole`